amethyst_core/src/float.rs

```diff
@@ -42,6 +44,30 @@ impl Float {
     }
 }
 
+impl PartialEq<f32> for Float {
+    fn eq(&self, other: &f32) -> bool {
+        (self.0 as f32).eq(other)
+    }
+}
+
+impl PartialEq<f64> for Float {
```

Comment:
This can have bad consequences. When `Float` is using `f32` internally, it will sill lead to comparisons being made with casts to `f64`s on both sides. This is because float literals default to f64 when multiple variants are available. I think we can have `PartialEq<f32>` though, as you have to cast and can't really do much than dropping a few bits of precission when comparing against low precision value in the first place.

Reply:
Or actually i'm not really sure about that. Whichever we pick ad "default", we are always suspectible to autocasting errors.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8477abe94ff6ac674d0a2d637e592f20
That playground shifts calculations to `f64` unintentionally, but it's equally easy to do this the other way around. The same issue persists with all `impl Into<Float>`
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7b624857eb01782b7ae6f7f6443ffa79

Reply:
soooo what do we do? ^^

Reply:
Actually I think this is fine. If it defaults to f64, it will get converted later to f32 internally. Sure, we are causing accidental float casts, but I don't see how to avoid that with the current state of type inference without totally removing the f64 impl. For example the From<f64>. That would prevent the cast, but would force users using f64 to cast to f32 and then back to f64 internally.